roles/otel_collector/defaults/main.yml

```diff
@@ -0,0 +1,10 @@
+otel_collector_version: "0.90.1"
+otel_collector_service_name: "otel-collector"
+otel_collector_type: contrib
+otel_collector_installation_dir: "/etc/otel-collector"
+otel_collector_config_dir: "/etc/otel-collector"
+otel_collector_config_file: "config.yaml"
+otel_collector_service_user: "otel"
+otel_collector_service_group: "otel"
+
+otel_collector_config: ""
```

Comment:
Maybe config can be split into main sections like exporters,receivers, etc? So they can be defined independently?

Reply:
yeah I did think breaking it down but sometimes users alread have a config file and its easier to copy paste the file rather than converting each component into something compatible with our role

Reply:
Consider this scenario: Ansible is used to manage fleet of hosts, sometimes large fleets. This would be common that default config file is distributed to all hosts, but some group of hosts require more specific overrides. With fractioning of a config , it would be easier to override just parts of the collectors' config. Similar to grafana-agent config you had, where you can define extra scrape_targets etc.

Reply:
Got it, But even in those cases the config would be passed to all installations right?
Like even in agent when I add scrape_targets for example, That peice of config is then present on all agent installations.

> it would be easier to override just parts of the collectors' config

I understand your ask though How would this look like, The user would still need to define each element in config right anyways?

Reply:
Alright, So we would just be adding these variables for

[Receivers](https://opentelemetry.io/docs/collector/configuration/#receivers) 
[Processors](https://opentelemetry.io/docs/collector/configuration/#processors) 
[Exporters](https://opentelemetry.io/docs/collector/configuration/#exporters) 
[Connectors](https://opentelemetry.io/docs/collector/configuration/#connectors)
Service
extensions